azure-storage/spring-cloud-starter-stream-sink-azure-blob/src/main/java/org/springframework/cloud/stream/app/azure/storage/sink/AzureBlobSinkConfiguration.java

```diff
@@ -0,0 +1,140 @@
+            ";AccountKey=" + this.properties.getAccountKey();
+
+        try {
+            // Setup the cloud storage account.
+            CloudStorageAccount account = CloudStorageAccount.parse(storageConnectionString);
+
+            LOG.info("getBlobService() : using account {}", this.properties.getAccountName());
+
+            // Create a blob service client
+            CloudBlobClient blobClient = account.createCloudBlobClient();
+            
+            // Get a reference to a container
+            // The container name must be lower case
+            CloudBlobContainer container = blobClient.getContainerReference(this.properties.getContainerName().toLowerCase());
+
+            LOG.info("getBlobService() : using container {}", this.properties.getContainerName());
+
+            if (this.properties.getAutoCreateContainer()) {
+                container.createIfNotExists();
+            }            
+
+            // Make the container public
+            if (this.properties.getPublicPermission()) {
+                LOG.info("getBlobService() : making container publically accessible");
+
+                // Create a permissions object
+                BlobContainerPermissions containerPermissions = new BlobContainerPermissions();
+
+                // Include public access in the permissions object
+                containerPermissions.setPublicAccess(BlobContainerPublicAccessType.CONTAINER);
+
+                // Set the permissions on the container
+                container.uploadPermissions(containerPermissions);
+            }
+
+            LOG.info("getBlobService() : using blob name {}", this.properties.getBlobName());
+            
+            if (this.properties.getAppendOnly()) {
+                this.blobService = container.getAppendBlobReference(this.properties.getBlobName());
+                if (this.properties.getOverwiteExistingAppend()) {
+                    ((CloudAppendBlob) blobService).createOrReplace();
+                }
+            }
+            else {
+                this.blobService = container.getBlockBlobReference(this.properties.getBlobName());
+            }
+        } catch (Exception e) {
+            // Log the stack trace.
+            LOG.error("getBlobService() : {}", e.getMessage());
+            //throw e;
+        }
+    }
+
+    @ServiceActivator(inputChannel=Sink.INPUT)
+    public void pushToAzureBlob(Message<?> message) throws MessagingException {
+        try {
+            // Upload the payload to the blob
+            if (this.properties.getAppendOnly()) {
+                ((CloudAppendBlob) blobService).appendText(message.getPayload().toString() + "\n");
+            }
+            else {
+                ((CloudBlockBlob) blobService).uploadText(message.getPayload().toString());
+            }
+
+        } catch (Exception e) {
+            // Log the stack trace.
+            LOG.error("pushToAzureBlob() : {}", e.getMessage());
```

Comment:
This solution doesn't look good.
Would be better just don't catch any exceptions and let them bubble to the caller.